integrations/qdrant/src/haystack_integrations/document_stores/qdrant/filters.py

```diff
@@ -141,34 +133,47 @@ def _build_nin_condition(self, key: str, value: List[models.ValueVariants]) -> m
             must_not=[
                 (
                     models.FieldCondition(key=key, match=models.MatchText(text=item))
-                    if isinstance(item, str) and " " not in item
+                    if isinstance(item, str) and " " in item
```

Comment:
can you explain this change?

Reply:
This fixes the previous faulty implementation.
If the string has a whitespace we do a [full-text match](https://qdrant.tech/documentation/concepts/filtering/#full-text-match) else a [keyword/number match](https://qdrant.tech/documentation/concepts/filtering/#match).